packages/compass-schema/src/components/export-schema-modal.tsx

```diff
@@ -104,6 +112,25 @@ const ExportSchemaModal: React.FunctionComponent<{
     [onChangeSchemaExportFormat]
   );
 
+  const handleSchemaDownload = useCallback(() => {
+    try {
+      if (!exportedSchema) return;
+      const blob = new Blob([exportedSchema], {
+        type: 'application/json',
+      });
+      const url = window.URL.createObjectURL(blob);
+      const link = document.createElement('a');
+      link.href = url;
+      link.download = filename || 'export.json';
+      link.click();
+      window.URL.revokeObjectURL(url);
+      onExportedSchema();
+    } catch (error) {
+      onSchemaExportFailed('download button clicked');
+      throw error;
+    }
+  }, [exportedSchema, filename, onSchemaExportFailed, onExportedSchema]);
```

Comment:
This whole thing looks like a store action to me that doesn't necessarily needs to be part of the UI: all this state is coming from the store, the tracking "stages" feel like a bit too much info to give UI control over, and we probably don't want to throw in the click handler in the UI even if it failed, this should probably either dispatch an action of its own if we're planning to store and map it to some UI or just logged and ignored if not